lib/dockerregistry/inventory_test.go

```diff
@@ -2049,6 +2049,133 @@ func TestRunChecks(t *testing.T) {
 	}
 }
 
+func TestImageSizeCheck(t *testing.T) {
+	var tests = []struct {
+		name      string
+		check     reg.ImageSizeCheck
+		manifests []reg.Manifest
+		expected  error
+	}{}
+
+	for _, test := range tests {
+		edges, _ := reg.ToPromotionEdges(test.manifests)
+		got := test.check.Run(edges)
+		err := checkEqual(got, test.expected)
+		checkError(t, err,
+			fmt.Sprintf("checkError: test: %v (Error Tracking)\n", test.name))
+	}
+}
+func TestImageRemovalCheck(t *testing.T) {
+	srcRegName := reg.RegistryName("gcr.io/foo")
+	destRegName := reg.RegistryName("gcr.io/bar")
+	destRC := reg.RegistryContext{
+		Name:           destRegName,
+		ServiceAccount: "robot",
+	}
+	srcRC := reg.RegistryContext{
+		Name:           srcRegName,
+		ServiceAccount: "robot",
+		Src:            true,
+	}
+	registries := []reg.RegistryContext{destRC, srcRC}
+
+	image1 := reg.Image{
+		ImageName: "foo",
+		Dmap: reg.DigestTags{
+			"sha256:000": {"0.9"}}}
+	image2 := reg.Image{
+		ImageName: "bar",
+		Dmap: reg.DigestTags{
+			"sha256:000": {"0.9"}}}
+
+	var tests = []struct {
+		name       string
+		check      reg.ImageSizeCheck
+		manifests  []reg.Manifest
+		imageSizes []int
```

Comment:
This is a little bit confusing if you have more than 1 image; why not use a map[Digest]int ?